src/web/tools/GestureHandlerOrchestrator.ts

```diff
@@ -33,47 +34,48 @@ export default class GestureHandlerOrchestrator {
   public removeHandlerFromOrchestrator(handler: IGestureHandler): void {
     this.gestureHandlers.splice(this.gestureHandlers.indexOf(handler), 1);
     this.awaitingHandlers.splice(this.awaitingHandlers.indexOf(handler), 1);
+    this.awaitingHandlersTags.delete(handler.getTag());
   }
 
   private cleanupFinishedHandlers(): void {
+    const handlersToRemove = new Set<IGestureHandler>();
+
     for (let i = this.gestureHandlers.length - 1; i >= 0; --i) {
       const handler = this.gestureHandlers[i];
 
-      if (!handler) {
-        continue;
-      }
       if (this.isFinished(handler.getState()) && !handler.isAwaiting()) {
-        this.gestureHandlers.splice(i, 1);
-
         this.cleanHandler(handler);
+        handlersToRemove.add(handler);
       }
     }
+
+    this.gestureHandlers = this.gestureHandlers.filter(
+      (handler) => !handlersToRemove.has(handler)
+    );
   }
 
   private hasOtherHandlerToWaitFor(handler: IGestureHandler): boolean {
-    let hasToWait = false;
-    this.gestureHandlers.forEach((otherHandler) => {
-      if (
-        otherHandler &&
+    const hasToWaitFor = (otherHandler: IGestureHandler) => {
+      return (
         !this.isFinished(otherHandler.getState()) &&
         this.shouldHandlerWaitForOther(handler, otherHandler)
-      ) {
-        hasToWait = true;
-        return;
-      }
-    });
+      );
+    };
 
-    return hasToWait;
+    return this.gestureHandlers.some(hasToWaitFor);
   }
 
   private shouldBeCancelledByFinishedHandler(
     handler: IGestureHandler
   ): boolean {
-    return this.gestureHandlers.some(
-      (otherHandler) =>
+    const shouldBeCancelled = (otherHandler: IGestureHandler) => {
```

Comment:
```suggestion
    const shouldBeCancelledBy = (otherHandler: IGestureHandler) => {
```

Reply:
Done in [145e9f7](https://github.com/software-mansion/react-native-gesture-handler/pull/2819/commits/145e9f710b298a064f7ff16ef34195826e3cb807).